### PR TITLE
Add kafka channel and related changes from upstream

### PR DESCRIPTION
**Pull request by pierDipi**
- Bring KafkaChannel reconciler
- Run KafkaChannel tests

I had to bring also the experimental trigger filters feature since the contract changes for the Channel migration was done after this feature.

See individual commits for details.

(Note: this doesn't really work yet)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,16 +28,16 @@ data-plane/**/build/
...
+/eventing-kafka-channel.yaml
+/eventing-kafka-sink.yaml
+
+/eventing-kafka-post-install.yaml
+
+/eventing-kafka-controller-prometheus-operator.yaml
+/eventing-kafka-source-prometheus-operator.yaml
+/eventing-kafka-channel-prometheus-operator.yaml
+/eventing-kafka-broker-prometheus-operator.yaml
+/eventing-kafka-sink-prometheus-operator.yaml
```
Comment (pierDipi) [inline .gitignore:43]: without `/` it ignores files in `openshift/release/artifacts` with the same name.

**Comment by pierDipi** [comment]
```
Error: timeout waiting for machineset  in namespace openshift-machine-api to scale up to 6 replicas
```
/test 49-e2e-aws-ocp-49

**Comment by pierDipi** [comment]
```
openshift/e2e-common.sh: line 127: KNATIVE_EVENTING_KAFKA_BROKER_POST_INSTALL: unbound variable
```
missing image dependency for conformance tests

**Comment by pierDipi** [comment]
```
openshift/e2e-common.sh: line 127: KNATIVE_EVENTING_KAFKA_BROKER_POST_INSTALL: unbound variable
```
missing image dependency for conformance tests

**Comment by pierDipi** [comment]
Fix https://github.com/openshift/release/pull/27976

**Comment by pierDipi** [comment]
Failed again with:
```
Error: timeout waiting for machineset  in namespace openshift-machine-api to scale up to 6 replicas
```

**Comment by matzew** [comment]
> Error: timeout waiting for machineset  in namespace openshift-machine-api to scale up to 6 replica

we have seen this before - it's sometimes related to load on the CI cluster and infra in general :disappointed:

**Comment by matzew** [comment]
```
 eventing_kafka_sink=openshift/release/artifacts/eventing-kafka-sink.yaml
+ sed -i -e 's|registry.ci.openshift.org/openshift/knative-.*:knative-eventing-kafka-broker-kafka-controller|registry.build02.ci.openshift.org/ci-op-hd3nwxtt/pipeline@sha256:bc9a9a1cc6055a56a61f4a8038c354db315014bdca0a7f3c1f6249ffe2ebaefa|g' openshift/release/artifacts/eventing-kafka-controller.yaml
+ sed -i -e 's|registry.ci.openshift.org/openshift/knative-.*:knative-eventing-kafka-broker-webhook-kafka|registry.build02.ci.openshift.org/ci-op-hd3nwxtt/pipeline@sha256:112f81fbe4fb4e24519781c028b868079c12077dc50c01880216fefca72cb24f|g' openshift/release/artifacts/eventing-kafka-controller.yaml
openshift/e2e-common.sh: line 127: KNATIVE_EVENTING_KAFKA_BROKE
```

there seems to be a real issue

**Comment by pierDipi** [comment]
No, it's the same issue as yesterday
```
openshift/e2e-common.sh: line 127: KNATIVE_EVENTING_KAFKA_BROKER_POST_INSTALL: unbound variable
```
missing image as dependency of 2 jobs `49-conformance-aws-ocp-49` and `410-e2e-aws-ocp-410`

**Comment by pierDipi** [comment]
Fix in https://github.com/openshift/release/pull/28016

**Comment by pierDipi** [comment]
`410-e2e-aws-ocp-410` passing now!

File: openshift/e2e-tests.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ export TEST_IMAGE_TEMPLATE="${EVENTING_KAFKA_BROKER_TEST_IMAGE_TEMPLATE}"
 
 env
 
-scale_up_workers || exit 1
+#scale_up_workers || exit 1
```
Comment (aliok) [inline openshift/e2e-tests.sh:13]: All good, just curious about this one.

Why does this fail?

Comment (pierDipi) [reply openshift/e2e-tests.sh:13]: it times out https://github.com/openshift-knative/eventing-kafka-broker/pull/221#issuecomment-1103819913.

Perhaps we just don't need more workers in this repository.

**Comment by matzew** [comment]
Awesome!

**Pull request merged**